qubesbuilder/cli/cli_base.py

```diff
@@ -59,12 +59,41 @@ def __init__(self, *args, **kwargs):
         self.debug = False
         self.list_commands = self.list_commands_for_help  # type: ignore
 
+        # Store global cleanup callbacks to run on interrupt or error
+        self._cleanup_callbacks: List[Callable[[], None]] = []
+
+    def add_cleanup(self, fn: Callable[[], None]):
+        """Register a function to run on interrupt or error."""
+        self._cleanup_callbacks.append(fn)
+
+    def cleanup(self):
+        """Call all the cleanup callbacks."""
+        for fn in self._cleanup_callbacks:
+            try:
+                fn()
+            except Exception as e:
+                QubesBuilderLogger.error(f"Cleanup callback failed: {e}")
```

Comment:
It's this one that needs printing traceback with --debug, not cancelation itself.